src/Compilers/CSharp/Portable/Compilation/InitializerSemanticModel.cs

```diff
@@ -275,5 +275,24 @@ protected override void AnalyzeBoundNodeNullability(BoundNode boundRoot, Binder
         {
             NullableWalker.AnalyzeWithoutRewrite(Compilation, MemberSymbol, boundRoot, binder, diagnostics, createSnapshots);
         }
+
+#nullable enable
+
+        protected override bool IsNullableAnalysisEnabled()
+        {
+            switch (MemberSymbol.Kind)
+            {
+                case SymbolKind.Field:
+                case SymbolKind.Property:
+                    return MemberSymbol.ContainingType is SourceMemberContainerTypeSymbol type &&
+                        type.IsNullableEnabledForConstructorsAndInitializers(useStatic: MemberSymbol.IsStatic);
+                case SymbolKind.Parameter:
+                    return SourceComplexParameterSymbol.GetDefaultValueSyntaxForIsNullableAnalysisEnabled(Root as ParameterSyntax) is { } value &&
+                        Compilation.IsNullableAnalysisEnabledIn(value);
+                default:
```

Comment:
The other `switch (MemberSymbol.Kind)` throws on the `default` case. Is there a reason we don't replicate that behavior here?